packages/compose/src/hooks/use-previous/index.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Use something's value from the previous render.
+ * Based on https://usehooks.com/usePrevious/.
+ *
+ * @param {any} value The value to track.
+ *
+ * @return {any} The value from the previous render.
```

Comment:
@aduth How can we tell typescript that the returned value is the same type as the provided one?

Reply:
I wasn't sure how to do it before, but I looked it up and apparently you can do it by using `@template`. I've updated the JSDoc to use generic template types now. Let me know if you're okay with the approach and I'll merge.

Reply:
Yeah `@template` is the correct way to do this. 👍 

https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html